demos/convolution.cpp

```diff
@@ -0,0 +1,66 @@
+/*******************************************************************************
+* Copyright 2009-2015 Juan Francisco Crespo Gal�n
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*   http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+******************************************************************************/
+
+#include "fx/ConvolverSound.h"
+#include "fx/ImpulseResponse.h"
+#include "util/StreamBuffer.h"
+#include "util/ThreadPool.h"
+#include "devices/DeviceManager.h"
+#include "devices/IDevice.h"
+#include "devices/IDeviceFactory.h"
+#include "devices/IHandle.h"
+#include "plugin/PluginManager.h"
+#include "file/File.h"
+#include "IReader.h"
+
+#include <iostream>
+#include <condition_variable>
```

Comment:
You don't need condition variables in this demo (and others). Please try to keep the number of includes as small as possible everywhere.